announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java

```diff
@@ -2540,6 +2659,25 @@ protected void readAnnouncementForm(RunData rundata, Context context, boolean ch
 			state.setTempAnnounceToGroups(null);
 		}
 
+		if (announceTo != null && announceTo.equals("roles")) {
```

Comment:
"roles".equals(announceTo), simple! Create a constant for this.

Reply:
What do you mean with create a constant to this? It is for the "roles" string or to the "announceTo.equals("roles")"

Reply:
ROLES_CONSTANT.equals(announceTo) or StringUtils.equalsIgnoreCase(announceTo, ROLES_CONSTANT)

Reply:
Ok. Perfect, thanks